assets/onboarding/features/installation-feedback.jsx

```diff
@@ -1,49 +1,74 @@
+import { useState, useEffect } from '@wordpress/element';
 import { List } from '@woocommerce/components';
 import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
 import FeatureDescription from './feature-description';
-import FeatureStatus, { LOADING_STATUS, ERROR_STATUS } from './feature-status';
+import FeatureStatus, {
+	INSTALLING_STATUS,
+	ERROR_STATUS,
+} from './feature-status';
+import useFeaturesPolling from './use-features-polling';
+
+const getStatus = ( status = INSTALLING_STATUS ) => status;
 
-/**
- * @typedef  {Object} Feature
- * @property {string} title          Feature title.
- * @property {string} excerpt        Feature excerpt.
- * @property {string} [link]         Feature link.
- * @property {string} [errorMessage] Error message.
- * @property {string} status         Feature status.
- */
 /**
  * Installation feedback component.
  *
  * @param {Object}    props
- * @param {Feature[]} props.features   Features list.
  * @param {Function}  props.onContinue Callback to continue to the next step.
+ * @param {Function}  props.onRetry    Callback to retry installations.
  */
-const InstallationFeedback = ( { features, onContinue } ) => {
-	const hasLoading = features.some(
-		( feature ) => feature.status === LOADING_STATUS
-	);
+const InstallationFeedback = ( { onContinue, onRetry } ) => {
+	const [ hasInstalling, setHasInstalling ] = useState( true );
+	const [ hasError, setHasError ] = useState( false );
 
-	const hasError = features.some(
-		( feature ) => feature.status === ERROR_STATUS
+	// Polling is active while some feature is installing.
+	const featuresData = useFeaturesPolling( hasInstalling );
+	const features = featuresData.options.filter( ( feature ) =>
+		featuresData.selected.includes( feature.slug )
 	);
 
+	// Update general statuses when features is updated.
+	useEffect( () => {
+		setHasInstalling(
+			features.some(
+				( feature ) => getStatus( feature.status ) === INSTALLING_STATUS
+			)
+		);
+
+		setHasError(
+			features.some(
+				( feature ) => getStatus( feature.status ) === ERROR_STATUS
+			)
+		);
+	}, [ features ] );
+
 	let actionButtons;
 
-	if ( hasLoading ) {
+	if ( hasInstalling ) {
 		actionButtons = (
 			<Button isPrimary className="sensei-onboarding__button">
 				{ __( 'Installing…', 'sensei-lms' ) }
 			</Button>
```

Comment:
We could add `isBusy` and `disabled` here to make it non-interactive.

Reply:
Looks good to me. https://github.com/Automattic/sensei/pull/3190/commits/fa1af5eac1de5b4445ca0d2847a4c9f3b86c24e9. It just is a little different than the design because it has the busy animation, but I think it looks better in this way. Make sense @donnapep ?

Reply:
Yes, this is what we're currently doing with the button state. In this case though, the animation looks to be resetting while extensions are installed. Is there an easy fix for that?

Reply:
Hum.. I checked and actually it's a problem that is happening because we're overriding the CSS `background-image`. I didn't dig so much because if we can use variables while working in this issue: https://github.com/Automattic/sensei/issues/3173, it will be fixed. I added [this comment](https://github.com/Automattic/sensei/issues/3173#issuecomment-635351116) to we check it while working in this issue, okay?